src/engines/html5/html5.js

```diff
@@ -246,6 +246,76 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     }
   }
 
+  /**
+   * this function runs on the players text tracks and checks if there is already a text track with the same language
+   * as the new one.
+   * @param {PKTextTrack} textTrack - the text track to check
+   * @returns {number} - the index of the text track with the same language (if there is any). and -1 if there isn't
+   * @public
+   */
+  getLanguageTrackIndex(textTrack: PKTextTrack): number {
+    const trackList = this._el.textTracks;
+    let index = -1;
+    for (let i = 0; i < trackList.length; i++) {
+      if (trackList[i].language === textTrack.language) {
+        index = i;
+        break;
+      }
+    }
+    return index;
+  }
+
+  /**
+   * adds a new text track element to the video element or set an existing one
+   * (when adding a text track with existing language to the video element it will remove all its cues)
+   * @param {PKTextTrack} textTrack - the playkit text track object to be added
+   * @returns {void}
+   * @public
+   */
+  addTextTrack(textTrack: PKTextTrack): TextTrack {
+    let domTrack;
+    const sameLanguageTrackIndex = this.getLanguageTrackIndex(textTrack);
+    if (sameLanguageTrackIndex > -1) {
+      domTrack = this._el.textTracks[sameLanguageTrackIndex];
+      domTrack.mode = 'hidden';
+      this._removeCues(domTrack);
+    } else {
+      domTrack = this._el.addTextTrack("subtitles", textTrack.label || textTrack.language, textTrack.language);
+    }
+    return domTrack;
+  }
+
+  /**
+   * remove cues from an HTML text track
+   * @param {TextTrack} textTrack - an HTML text track
+   * @returns {void}
+   * @private
+   */
+  _removeCues(textTrack: TextTrack): void {
```

Comment:
Maybe consider this as an API?

Reply:
don't think it's currently a use case. it is used to clean the text track internally. we can expose it in the future, if needed.